core/src/main/java/ai/timefold/solver/core/impl/solver/DefaultSolverFactory.java

```diff
@@ -116,7 +117,10 @@ public <Score_ extends Score<Score_>> ScoreDirectorFactory<Solution_, Score_> ge
                     metricsRequiringConstraintMatchSet);
         }
 
+        var valueRangeState = new ValueRangeState<Solution_>();
```

Comment:
You should be able to reuse this instance in selectors - it should eliminate the need to worry about what's in which range. You simply ask this class, and it tells you yes/no.

If necessary, we can also add a method to expose the range, so that you don't need to create your own instances.